instrumentation/java-http-server/library/src/main/java/io/opentelemetry/instrumentation/httpserver/internal/JdkHttpServerAttributesGetter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.httpserver.internal;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpsExchange;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpServerAttributesGetter;
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+enum JdkHttpServerAttributesGetter
+    implements HttpServerAttributesGetter<HttpExchange, HttpExchange> {
+  INSTANCE;
+
+  @Override
+  public String getHttpRequestMethod(HttpExchange exchange) {
+    return exchange.getRequestMethod();
+  }
+
+  @Override
+  public String getUrlScheme(HttpExchange exchange) {
+    return exchange instanceof HttpsExchange ? "https" : "http";
+  }
+
+  @Override
+  public String getUrlPath(HttpExchange exchange) {
+    String fullPath = exchange.getRequestURI().toString();
+    int separatorPos = fullPath.indexOf('?');
+    return separatorPos == -1 ? fullPath : fullPath.substring(0, separatorPos);
+  }
+
+  @Nullable
+  @Override
+  public String getUrlQuery(HttpExchange exchange) {
+    String fullPath = exchange.getRequestURI().toString();
+    int separatorPos = fullPath.indexOf('?');
+    return separatorPos == -1 ? null : fullPath.substring(separatorPos + 1);
+  }
+
+  @Override
+  public List<String> getHttpRequestHeader(HttpExchange exchange, String name) {
+    return exchange.getRequestHeaders().getOrDefault(name, Collections.emptyList());
+  }
+
+  @Nullable
+  @Override
+  public Integer getHttpResponseStatusCode(
+      HttpExchange exchange, @Nullable HttpExchange res, @Nullable Throwable error) {
+    int status = exchange.getResponseCode();
+    if (status > 1) {
+      return status;
+    }
+    return null;
+  }
+
+  @Override
+  public List<String> getHttpResponseHeader(
+      HttpExchange exchange, @Nullable HttpExchange res, String name) {
+    return exchange.getResponseHeaders().getOrDefault(name, Collections.emptyList());
+  }
+
+  @Override
+  public String getHttpRoute(HttpExchange exchange) {
+    return exchange.getHttpContext().getPath();
```

Comment:
Route must have a low cardinality. Often it is not possible to get route for a low level server framework. Just return `null` here.

Reply:
I don't follow, `exchange.getHttpContext().getPath()` gives you the configured path for the route, not what is currently matched

Reply:
that is, for 
```java
    HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
    server.createContext("/path/" ctx ->{});
```

and send a request to `/path/something`, you'll get `/path/` from `exchange.getHttpContext().getPath()`